<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Integrative Genomics Viewer</title>
    <meta charset="utf-8" />
    <meta name="author" content="Rockefeller University, Bioinformatics Resource Centre" />
    <script src="libs/header-attrs-2.29/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="metropolisCustom.css" type="text/css" />
    <link rel="stylesheet" href="metropolis-fontsCustom.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Integrative Genomics Viewer
]
.author[
### Rockefeller University, Bioinformatics Resource Centre
]
.date[
### <a href="https://rockefelleruniversity.github.io/IGV_course/" class="uri">https://rockefelleruniversity.github.io/IGV_course/</a>
]

---

# IGV

---
## Overview
.pull-left[
- Introduction to IGV.
    - What is IGV.
    - How to run IGV Desktop.
- Navigating IGV.
    - The IGV user interface.
    - Moving around genomes.
- Loading and visualising data.
    - Genome information and annotation.
    - User supplied data.
    - Sample information.
    - External data.
- Displaying genomics data
    - Basic visualisation.
    - Data dependent visualisation.
]
.pull-right[
- IGV web
  - How to run IGV web
- Navigating IGV web
- Loading and visualising data
- Sharing web sessions
]

---
## What is IGV?

- Created by the Broad institute.
- Desktop Genome browser.
    - Visualises genomic data (expression, ChIP, resequencing, multiple alignment, shRNA).
    - Handles most common genomic data types.
    - Local java application
    - No dependence on server.
    - Loads data locally or from URL, consumes local memory and CPU.

---
## How to get IGV?

- Desktop, Web app and Juicebox
  - Desktop - Runs locally on desktop using Java
  - Web app - Runs locally in web browser using JavaScript
  - Juicebox - For HiC like data, runs locally on desktop using Java

&lt;div align="center"&gt;
&lt;img src="imgs/broad2.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

http://www.broadinstitute.org/software/igv/download


---
## How to run IGV Desktop?

.pull-left[
- Download to computer (recommend those including Java).
  - Runs locally out of the box.
- Archived versions available.
  - In case you liked functionality of previous version better.
- Snapshots of development versions available.
  - These may be unstable and/or have features which are not included in release version.
]
.pull-right[
&lt;div align="center"&gt;
&lt;img src="imgs/binary2.png" alt="igv" height="400" width="300"&gt;
&lt;/div&gt;
]

---
## IGV GUI

![](imgs/igvGUI2.png)

---
## IGV GUI

.pull-left[
![](imgs/igvGUIh
.png)
  ]
.pull-right[
- Sample information panel &lt;1&gt;
- Genome Navigation panel &lt;2&gt;
- Data panel &lt;3&gt;
- Attribute panel &lt;4&gt;
  ]
  
---
## Menu bar

![](imgs/menubar2.png)

- File - Load data/sample information.
- Genomes - Load and manage genomes.
- View -  Display preferences.
- Tracks - Group/sort/filter data tracks.
- Regions - Create region/gene lists.
- Tools -  Access to Integrated tools (IGVtools/Bedtools).

---
## Moving around genomes
- Cytoband selection and zooming
- Scrolling
- Selection of region of interest

&lt;div align="center"&gt;
&lt;img src="imgs/moving.png" alt="igv"&gt;
&lt;/div&gt;

---
## Moving around genomes

![](imgs/moving.png)

---
## Zoom


.pull-left[
![](imgs/zoom1.png)
  ]
.pull-right[
![](imgs/zoom2.png)
  ]
  
---
## Scrolling

![](imgs/scrolling.png)

---
## Jump to region

.pull-left[
![](imgs/jmp1.png)
  ]
.pull-right[
![](imgs/jmp2.png)
  ]

---

## “Bookmarking” regions of interest

- Regions may be added to “Regions of interest”.
 
- These act as bookmarks for areas of particular interest.
 
- Bookmarks can be added by:
    - Bookmarking visible window.
    - Selecting region within window.

---
## Bookmarking - Visible window


.pull-left[
![](imgs/book.png)
  ]
.pull-right[
![](imgs/book2.png)
  ]

---

## Bookmarking
&lt;div align="center"&gt;
&lt;img src="imgs/book3.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;

---

## Bookmarking

- Bookmarks may be created quickly by selecting "region of interest" button.
- Once selected just point-and-click where you want each edge to be.

&lt;div align="center"&gt;
&lt;img src="imgs/book4.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;


---

## Sequences from regions of interest
- A useful feature of "regions of interest" is to retrieve the sequence. 
- You can even directly BLAT sequences.

&lt;div align="center"&gt;
&lt;img src="imgs/seq1.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;

---

## Sequences from regions of interest
- BLAT results present as table of hits.

&lt;div align="center"&gt;
&lt;img src="imgs/seq2.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;


---

## Viewing sequences

- At a predefined resolution, sequence information becomes available.

&lt;div align="center"&gt;
&lt;img src="imgs/seq3.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;


---
## Viewing sequences

- Strand of sequence can be altered.
- Differing translation tables can be selected.

&lt;div align="center"&gt;
&lt;img src="imgs/seq4.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;


---
## Gene Lists
- When you do not have a region of interest but a group of genes you can use Gene Lists ("Regions" &gt; "Gene Lists"). 
- You create your own list or use one of the curated lists. 

&lt;div align="center"&gt;
&lt;img src="imgs/GL1.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;

---
## Gene Lists
Using gene lists you can visualise several genes concurrently.

&lt;div align="center"&gt;
&lt;img src="imgs/GL2.png" alt="igv" height="400" width="600"&gt;
&lt;/div&gt;


---
class: inverse, center, middle

# Loading data in IGV

&lt;html&gt;&lt;div style='float:left'&gt;&lt;/div&gt;&lt;hr color='#EB811B' size=1px width=720px&gt;&lt;/html&gt; 

---
## Loading Genome Information

- Many genomes can be selected from dropdown.

.pull-left[
![](imgs/lgi_1.png)
  ]
.pull-right[
![](imgs/lgi_2.png)
  ]


---
## Loading Genome Information



.pull-left[
- Genomes not included may be downloaded from repository
  - Users can select whether to download sequence/annotation or use version hosted online for many genomes.
  ]

.pull-right[
![](imgs/lgi3_2_2.png)
  ]

---

## Loading Genome Information

- For supported genomes, gene positions are automatically included in “feature” panel.
- Additional gene positions can be loaded into IGV in gff format

&lt;div align="center"&gt;
&lt;img src="imgs/moving.png" alt="igv" height="400" width="800"&gt;
&lt;/div&gt;

---
## Loading Sample data
- Most common formats can be loaded into IGV through file menu
- Acceptable data formats include:-
    - BED (.bed)
    - BAM and index (.bam with .bai/.bam.bai)
    - BigWig (.bw)
    - BedGraph/Wig (.bedGraph, .wig)
    - And many more...
    
- For a full list of compatible file formats check out the [IGV website](https://software.broadinstitute.org/software/igv/FileFormats)

---
## Loading Sample Metadata

.pull-left[
- IGV allows the inclusion of information on samples.
- Sample information is included in sample information panel.
]
.pull-right[
![](imgs/lsm.png)
]
---
## Loading Sample Metadata

Example Sample information file
http://www.broadinstitute.org/igvdata/exampleFiles/exampleSampleInfo.txt

&lt;div align="center"&gt;
&lt;img src="imgs/smdT.png" alt="igv" height="400" width="800"&gt;
&lt;/div&gt;


---
## Sample Metadata with TrackTables

- [Tracktables](https://bioconductor.org/packages/release/bioc/html/tracktables.html) is a R/Bioconductor package.
- Creates HTML reports with sample information for use with IGV.

&lt;div align="center"&gt;
&lt;img src="imgs/tracktables.png" alt="igv" height="400" width="500"&gt;
&lt;/div&gt;

---
## Creating SampleMetadata in excel

The most straightforward way to build is most often to use excel/openoffice.

- Create an ID column containing names of files you want to load.
- Add additional columns of metadata
- Export file as a tab separated file for loading into IGV.

&lt;div align="center"&gt;
&lt;img src="imgs/excelsamplemeta.png" alt="igv" height="400" width="500"&gt;
&lt;/div&gt;

---
## Using sample information

- Sample information can include discrete and continuous.
 
    - Can be used to “sort” and “filter” tracks.
 
    - Can split tracks across panels by “group”


---
## Loading Genome Information

.pull-left[
&lt;div align="center"&gt;
&lt;img src="imgs/tt1.png" alt="igv" height="150" width="500"&gt;
&lt;/div&gt;
  ]
.pull-right[
&lt;div align="center"&gt;
&lt;img src="imgs/tt2.png" alt="igv" height="400" width="500"&gt;
&lt;/div&gt;
  ]

---
## Loading external data and annotation
 ** Load data from a URL ** 
- As with UCSC, IGV supports data hosted on external servers.
- Data accessible from a URL such as HTTP and FTP can be loaded using "File" &gt; “Load from URL”.

---
## Load data from a URL
Try: "https://github.com/ThomasCarroll/tracktables-Data/raw/master/H3K4me3_Example.bw"

.pull-top[
![](imgs/ldu1.png)
  ]
.pull-bottom[
![](imgs/ldu2.png)
  ]


---
## Loading external data and annotation
** Load data from a server (IGV/Encode servers)** 

- Unlike UCSC, IGV comes with few external tracks.
 
 
- External tracks (relevant to the genome) can be loaded from the IGV server or Encode-IGV server.

---
## Load data from a server (IGV/Encode)

.pull-left[
![](imgs/ls1.png)
  ]
.pull-right[
![](imgs/ls_2.png)
  ]

---
## Grabbing encode data directly
- Encode data can be downloaded from UCSC.
    - http://genome.ucsc.edu/ENCODE/

- This however does not come with sample information provided through IGV interface.

---
class: inverse, center, middle

# Viewing Data

&lt;html&gt;&lt;div style='float:left'&gt;&lt;/div&gt;&lt;hr color='#EB811B' size=1px width=720px&gt;&lt;/html&gt; 

---
## Viewing data

- IGV associates common file formats with default display methods.
 
- Most of the time IGV will make a sensible choice how we wish to display data.

---
## Accepted formats and default display.

- Information on accepted file formats and default display can be found at:

 http://www.broadinstitute.org/software/igv/RecommendedFileFormats
 
 &lt;img src="imgs/af.png" alt="igv" height="400" width="700"&gt;

---
## Bed/bigBed

.pull-left[
- Basic
- Tab-delimited
- Chrom,Start,End
- Bed6
- bigBed (recommended)
  ]
.pull-right[
 &lt;img src="imgs/bbed.png" alt="igv" height="400" width="700"&gt;

  ]
---
## Wig, BedGraph and BigWig

.pull-left[
- Wig/bedGraph require high memory load.
- Recommended format is bigWig.
  ]
.pull-right[
 &lt;img src="imgs/wigBg.png" alt="igv" height="400" width="700"&gt;

  ]
---
## BAM alignment files
.pull-left[
- BAM files contain alignment information.
- Require an accompanying .bai index file for display.
  ]
.pull-right[
 &lt;img src="imgs/bam1.png" alt="igv" height="400" width="700"&gt;

  ]

---
## Finer control of display

- IGV allows for customization of track display.
 
- "View" &gt; "Preferences"
 
- Select track (right click)

---
## Display preferences

** General **

 &lt;div align="center"&gt;
&lt;img src="imgs/dpg.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## Display preferences

** Tracks **

 &lt;div align="center"&gt;
&lt;img src="imgs/dpt.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## Display preferences

** Alignments **

 &lt;div align="center"&gt;
&lt;img src="imgs/dpa.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## Track display options

** Graph/interval files**
- Track colour/appearance

 &lt;div align="center"&gt;
&lt;img src="imgs/tdg.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;
---
## Track display options

** Graph/interval files**
- Graph type

 &lt;div align="center"&gt;
&lt;img src="imgs/tdg2.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;
---
## Track display options
**Data Scaling**

 &lt;div align="center"&gt;
&lt;img src="imgs/tds.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;
---
## Track display options


Autoscaling adjusts to track's visible signal maximum.

 &lt;div align="center"&gt;
&lt;img src="imgs/scaling.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;
---
class: inverse, center, middle

# Some more useful IGV features

&lt;html&gt;&lt;div style='float:left'&gt;&lt;/div&gt;&lt;hr color='#EB811B' size=1px width=720px&gt;&lt;/html&gt; 

---

## IGV displays splicing

 &lt;div align="center"&gt;
&lt;img src="imgs/splicing.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;


---
## IGV's Sashimi plots

(https://software.broadinstitute.org/software/igv/Sashimi)

 &lt;div align="center"&gt;
&lt;img src="imgs/shashimi.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;


---
## IGV - Overlay tracks

 &lt;div align="center"&gt;
&lt;img src="imgs/overlay.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## Overlay tracks

 &lt;div align="center"&gt;
&lt;img src="imgs/overlay2.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## Track summarisation

IGV can be used to combine tracks

 &lt;div align="center"&gt;
&lt;img src="imgs/combineDT.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## Sum of two tracks

 &lt;div align="center"&gt;
&lt;img src="imgs/combineDT2.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## Create new summed track
There are also options for subtract, divide and multiply.
 &lt;div align="center"&gt;
&lt;img src="imgs/combineDT3.png" alt="igv" height="400" width="400"&gt;
&lt;/div&gt;

---
## IGVtools
- [IGVTools](https://software.broadinstitute.org/software/igv/igvtools) can be used to post-process genomics data.
- Includes indexing, sorting and genome graph creation.
.pull-left[
&lt;img src="imgs/itools.png" alt="igv" height="400" width="400"&gt;
]
.pull-right[
&lt;img src="imgs/itools2.png" alt="igv" height="200" width="400"&gt;
]

---
## Saving images of tracks
- "File" &gt; "Save Image"
- You can easily save images from IGV to vector based svg format. This means you can further customize your tracks and assemble them into figures using [Inkscape](https://inkscape.org/) or Illustrator. 

---
## Saving your session
- If you want to revisit your tracks at a later date, you can save the entire session. 
- When you restart later on you just have to reopen the session and you will be right back where you were. 
- "File" &gt; "Save Session"

---
## Where to get help?

- http://www.broadinstitute.org/igv/UserGuide
 
- https://groups.google.com/forum/#!forum/igv-help





    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
